api/cmd/portainer/main.go

```diff
@@ -589,7 +591,14 @@ func buildServer(flags *portainer.CLIFlags) portainer.Server {
 		log.Fatalf("failed loading edge jobs from database: %v", err)
 	}
 
-	applicationStatus := initStatus(instanceID)
+	applicationStatus := initStatus(instanceID, *flags.DemoEnvironment)
+	if applicationStatus.DemoEnvironment {
```

Comment:
@zees-dev @chiptus 
personally not a big fan of lagging pieces of application state around the app and passing references to it to different services. What if we'd instead capture it inside a service of some sort that would incapsulate the state and provide a few methods to check demo conditions?